### ISSUE TITLE
zoom does not work (on windows)

### ISSUE THREAD

**Issue by mmagnuski** [JuliaGraphics/Immerse.jl#16]
I just installed Julia 0.4 and went on to try out Immerse. However, in the first example from the readme I encounter a problem with the zoom button. After selecting a region of the plot like this:
![img02](https://github-cloud.s3.amazonaws.com/assets/8452354/10565483/5cd121be-75d1-11e5-9515-29b9256266dc.png)

The plot is updated to:
![img03](https://github-cloud.s3.amazonaws.com/assets/8452354/10565486/6ee83950-75d1-11e5-9e6a-ce280c010003.PNG)

If I select a small enough fragment of the plot the x and y axes change but the zoom is not performed:
![img04](https://github-cloud.s3.amazonaws.com/assets/8452354/10565490/875cf476-75d1-11e5-92d5-70672de650a9.PNG)

Here is my `versioninfo()`:
```
Julia Version 0.4.0
Commit 0ff703b* (2015-10-08 06:20 UTC)
Platform Info:
  System: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-3770 CPU @ 3.40GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.3
```

**Comment by timholy** [comment]
I don't think it's just windows; I'd guess recent changes to Gadfly broke this. I'm a bit swamped right now, but if you're desperate you could try checking out an older version of Gadfly.

**Comment by tbreloff** [comment]
Agreed with Tim.  If I had a guess, it has to do with a recent change in Gadfly that changed the purpose of the min/max coordinates.  `Coord.Cartesian` should give you a forced range, while setting the scales (`Scale.x_continuous`, for example) will "suggest" a range, but Gadfly will still pick what range to show.  @dcjones mentioned the reasoning in an issue somewhere but I couldn't find it just now. 

Anyways I suspect [this code](https://github.com/JuliaGraphics/Immerse.jl/blob/master/src/display_gadfly.jl#L374-L379) no longer works as expected.

**Comment by mmagnuski** [comment]
Thanks! I'm definitely not desperate, I just wanted to play around with Immerse and noticed this so I thought I will let you know. 
Good to know its not only Windows, I was just following a simple heuristic: if something breaks that might be windows ;)

**Comment by dcjones** [comment]
I guess I rather unceremoniously broke this. I'll try to fix it later, and in the meantime you can checkout on older version of Gadfly (v0.3.16 should still work).

**Comment by timholy** [comment]
@dcjones, no sweat; it's hard to read the mind of other packages that mess around inside the internals of yours :smile:. Presumably the sensible thing is for me to write Gadfly tests that exercise the functionality that Immerse needs.

**Comment by tbreloff** [comment]
Same here @timholy.  Except that I'm messing around in the internals of **both** Immerse and Gadfly :)

**Comment by timholy** [comment]
Ever feel like this?

![image](https://cloud.githubusercontent.com/assets/1525481/10590617/62747c64-7677-11e5-9276-bacc52f38f97.png)

**Comment by timholy** [comment]
See https://github.com/JuliaLang/METADATA.jl/pull/3906. It's a little unsatisfying but the best I have time for.

My one "real" comment is that it would be good to tag Gadfly versions a little more frequently, especially right before merging major work like the new pan/zoom functionality. 0.3.16 still contains an ocean of deprecation warnings, and it would have been good to be able to use a version that fixed those but did not include the later breaking work.

**Comment by tbreloff** [comment]
True @timholy .  On that note it would be ideal if master didn't have much "breaking work" on it either.  In my mind, master is the "mostly stable and preparing for the next tag" branch.  I try to do all my dev work on another branch, and then merge into master when it's ready.  There are lots of people that track master with some expectation of things continuing to work (even though that's probably naive).

**Comment by timholy** [comment]
It's hard because in a project with many dependencies, tagging becomes a big deal. I will probably personally move towards a model in which branches frequently customize the `.travis.yml` script to check out specific branches of other projects. But that's a little dangerous because it's easy to forget to remove those hacks later.

**Comment by timholy** [comment]
This will be fixed by https://github.com/JuliaGraphics/Immerse.jl/pull/47

**Comment by timholy** [comment]
Fixed by #47.

**Comment by mmagnuski** [comment]
:+1:
